wgpu-core/src/command/compute.rs

```diff
@@ -851,6 +889,211 @@ impl Global {
+                }
+                ArcComputeCommand::DispatchIndirect { ref buffer, offset } => {
+                    // if there is no pipeline set, don't inject the validation commands as we will error anyway
+                    if let Some(original_pipeline) = current_pipeline.clone() {
+                        // validate some buffer properties that won't be validated later
+                        check_buffer_usage(
+                            buffer.as_info().id(),
+                            buffer.usage,
+                            wgt::BufferUsages::INDIRECT,
+                        )
+                        .map_pass_err(scope)?;
+
+                        if offset % 4 != 0 {
+                            return Err(ComputePassErrorInner::UnalignedIndirectBufferOffset(
+                                offset,
+                            ))
+                            .map_pass_err(scope);
+                        }
+
+                        let end_offset =
+                            offset + mem::size_of::<wgt::DispatchIndirectArgs>() as u64;
+                        if end_offset > buffer.size {
+                            return Err(ComputePassErrorInner::IndirectBufferOverrun {
+                                offset,
+                                end_offset,
+                                buffer_size: buffer.size,
+                            })
+                            .map_pass_err(scope);
+                        }
+
+                        let (validation_pipeline_id, validation_bgl_id) = self
+                            .device_get_or_create_dispatch_indirect_validation_pipeline::<A>(
+                                device_id,
+                            )
+                            .map_err(ComputePassIndirectValidationError::ValidationPipeline)
+                            .map_pass_err(scope)?;
+
+                        let (dst_buffer_id, error) = self.device_create_buffer::<A>(
+                            device_id,
+                            &crate::resource::BufferDescriptor {
+                                label: None,
+                                size: 4 * 3,
+                                usage: wgt::BufferUsages::INDIRECT | wgt::BufferUsages::STORAGE,
+                                mapped_at_creation: false,
+                            },
+                            None,
+                        );
+                        if let Some(error) = error {
+                            return Err(ComputePassIndirectValidationError::Buffer(error))
+                                .map_pass_err(scope)?;
+                        }
+
+                        // align offset to min_storage_buffer_offset_alignment
+                        let aligned_offset = offset
+                            - offset % device.limits.min_storage_buffer_offset_alignment as u64;
+
+                        let (bind_group_id, error) = self.device_create_bind_group::<A>(
```

Comment:
When do these bind groups/buffers get dropped? As it looks now they will exist forever.

Reply:
That's a good point, I was thinking it will happen eventually since we use `Arc`s but I missed we still have the registry and `Id`s 😅. I think what I have to do is just remove the id from the registry.

Reply:
Resolving as this will no longer be relevant once I update the patch with the new implementation strategy.